src/dbwrapper.h

```diff
@@ -178,10 +184,10 @@ class CDBWrapper
     friend const std::vector<unsigned char>& dbwrapper_private::GetObfuscateKey(const CDBWrapper &w);
 private:
     //! custom environment this database is using (may be nullptr in case of default environment)
-    leveldb::Env* penv;
+    std::unique_ptr<leveldb::Env> penv;
```

Comment:
nit: Maybe rename to m_env while touching all lines? Maybe split the penv change into a separate commit for easier review?

Reply:
renamed and split off into separate commit

Reply:
I didn't functionally change the code and now there is a TSAN failure

Reply:
I couldn't make any sense of it, so reset it, but it is still here for anyone to look at: https://cirrus-ci.com/task/5612886578954240?logs=ci#L4869

Reply:
I think this occurs and is unrelated to this PR because:
```
main thread:
* AppInitMain
  * ChainstateManager::ActiveTip()
    * CChain::Tip() -- accesses `vChain` without a lock (read)
```

```
b-loadblk thread:
* ThreadImport
  * ActivateBestChain called (src/node/blockstorage.cpp:883)
    * ActivateBestChainStep
      * ConnectTip
        * SetTip
          * vChain resize (write)
```